samples/general/mobile_nerf_rayquery/mobile_nerf_rayquery.cpp

```diff
@@ -0,0 +1,1235 @@
+
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 0.01f, 256.0f);
+
+	// Each models may have submodels
+	int models_entry = 0;
+	for (int model_index = 0; model_index < num_models; model_index++)
+	{
+		int num_sub_model = models[models_entry].sub_model_num;
+
+		for (int sub_model_index = 0; sub_model_index < num_sub_model; sub_model_index++)
+		{
+			load_scene(model_index, sub_model_index, models_entry);
+			create_texture(model_index, sub_model_index, models_entry);
+			create_static_object_buffers(models_entry);
+			create_bottom_level_acceleration_structure(models_entry);
+			models_entry++;
+		}
+	}
+
+	create_top_level_acceleration_structure();
+	create_uniforms();
+	create_pipeline_layout();
+	create_descriptor_pool();
+	create_descriptor_sets();
+	prepare_pipelines();
+	build_command_buffers();
+
+	prepared = true;
+	LOGI("Prepare Done!");
+	return true;
+}
+
+void MobileNerfRayQuery::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceBufferDeviceAddressFeaturesKHR, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_BUFFER_DEVICE_ADDRESS_FEATURES, bufferDeviceAddress);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceAccelerationStructureFeaturesKHR, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ACCELERATION_STRUCTURE_FEATURES_KHR, accelerationStructure);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceRayQueryFeaturesKHR, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_RAY_QUERY_FEATURES_KHR, rayQuery);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceDescriptorIndexingFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_FEATURES_EXT, shaderUniformBufferArrayNonUniformIndexing);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceDescriptorIndexingFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_FEATURES_EXT, shaderSampledImageArrayNonUniformIndexing);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceDescriptorIndexingFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_FEATURES_EXT, shaderStorageBufferArrayNonUniformIndexing);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceDescriptorIndexingFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_FEATURES_EXT, runtimeDescriptorArray);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceDescriptorIndexingFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_FEATURES_EXT, descriptorBindingVariableDescriptorCount);
+	REQUEST_REQUIRED_FEATURE(gpu, VkPhysicalDeviceScalarBlockLayoutFeaturesEXT, VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SCALAR_BLOCK_LAYOUT_FEATURES_EXT, scalarBlockLayout);
+}
+
+void MobileNerfRayQuery::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
+	draw();
+
+	update_uniform_buffer();
+}
+
+void MobileNerfRayQuery::read_json_map()
+{
+	std::string assetBase = vkb::fs::path::get(vkb::fs::path::Type::Assets);
+	LOGI("Base assets path: {}", assetBase);
+
+#if defined(NERF_JSON_FILE)
```

Comment:
is the JSON format/options documented? maybe it could be explained in the README? wondering if the code could be simplified to avoid duplication between combo and non-combo modes?

Reply:
Answered below, happy to address in a future PR